modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataInputDelegate.java

```diff
@@ -136,14 +138,16 @@ NamedTypeResolver getParameterResolver(String parameterName) {
           .withReason(NULL_TYPE_ERROR)
           .onParameter(parameter.getName());
       return failure(parameter.getType(), failure);
+    } catch (ConnectionException e) {
+      return failure(newFailure(e).withMessage("Failed to establish connection: " + ExceptionUtils.getMessage(e))
+          .withFailureCode(CONNECTION_FAILURE).onKeys());
```

Comment:
shouldn't this be onComponent()? since we are resolving input metadata? I think the onKeys() should be used only when resolving metadata keys.

Reply:
Good catch. Yes, it should be onComponent().